### ISSUE TITLE
Add minutes/seconds support to Angle

### ISSUE THREAD

**Issue by ianmackenzie** [ianmackenzie/elm-units#21]
At a minimum:

```elm
Angle.minutes : Float -> Angle
Angle.inMinutes : Angle -> Float
Angle.seconds : Float -> Angle
Angle.inSeconds : Angle -> Float
```

Perhaps also some support for compound definitions, something like:

```elm
Angle.fromRecord : { degrees : Float, minutes : Float, seconds : Float } -> Angle
Angle.toRecord : Angle -> { degrees : Float, minutes : Float, seconds : Float }
```

Or perhaps:

```elm
Angle.degreesMinutesSeconds : ( Float, Float, Float ) -> Angle
Angle.inDegreesMinutesSeconds : Angle -> ( Float, Float, Float )
```

For the compound definitions, how exactly should negative angles be converted to degrees/minutes/seconds?

**Comment by ianmackenzie** [comment]
A couple more thoughts from musing about this on the bus:

* Degree and minute values could be represented as ints instead of floats
* Inferring angle sign from sign of the degree value is tempting, but won't work since the degree value may be zero!

**Comment by ianmackenzie** [comment]
Another potential API:

```elm
type Sign
    = Positive
    | Negative

fromDms : { sign : Sign, degrees : Int, minutes : Int, seconds : Float } -> Angle

toDms : Angle -> { sign : Sign, degrees : Int, minutes : Int, seconds : Float }
```

which would be used as

```elm
angle =
    Angle.fromDms
        { sign = Angle.Positive
        , degrees = 45
        , minutes = 30
        , seconds = 0
        }

{ sign, degrees, minutes, seconds } =
    Angle.toDms angle
```

Potential drawback: could be easy to destructure as

```elm
{ degrees, minutes, seconds} =
    Angle.toDms angle
```

and forget to check for sign.

**Comment by ianmackenzie** [comment]
Implemented in 447c3c53cca69e83ab63ccc445b41b407c539a5f.
